lib/ca_testing/extensions/selenium/webdriver/logs.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module Selenium
+  module WebDriver
+    class Logs
+      # @return [Hash]
+      #
+      # Perform the regular log fetch from Selenium, but then store the logs inside a Hash
+      def get(type)
+        cached_logs[type] = @bridge.log type
+      end
+
+      # @return [Integer]
+      #
+      # Write the logs to a filepath (If provided), or default file path (Configured).
+      def write_log_to_file(type, file = default_file_path)
+        get(type) unless cached_logs.key?(type)
+
+        cached_logs[type].each do |log_entry|
+          if file == $stdout
+            file << log_entry
+          else
+            IO.write(file, log_entry)
+          end
+        end
+      end
+
+      private
+
+      def default_file_path
+        CaTesting.chrome_log_path.tap { |path| raise "Set the path to store logs using CaTesting.chrome_log_path=" unless path }
```

Comment:
I don't think you need to use tap for this. 
I would set that string to a constant then write something like:
`raise SET_PATH_ERROR unless CaTesting.chrome_log_path.present?`

Reply:
The tap is just to check there's a value provided (If it's nil, then the execution will error).

Reply:
Okay no problem, I gave that suggestion more as it was a lot of characters on one line. Is it possible to at least put the sentence in a constant?

Reply:
@luke-hill but isn't Jason's form, or my own preference (because I'm still a Perl-head at heart):

```ruby
CaTesting.chrome_log_path or raise "Set the path to store logs using CaTesting.chrome_log_path="
```

simpler? There's really no benefit to using `#tap` here, which is normally for if you need the value of something but need to change its contents before you assign/use it.